Sources/SWBTaskExecution/TaskActions/SwiftCachingMaterializeKeyTaskAction.swift

```diff
@@ -69,22 +69,16 @@ public final class SwiftCachingMaterializeKeyTaskAction: TaskAction {
             guard let cas = try swiftModuleDependencyGraph.getCASDatabases(casOptions: taskKey.casOptions, compilerLocation: taskKey.compilerLocation) else {
                 throw StubError.error("unable to use CAS databases")
             }
-            let jobTaskIDBase: UInt = 0
-            if let cachedComp = try cas.queryLocalCacheKey(taskKey.cacheKey) {
-                if try requestCompilationOutputs(cachedComp,
-                                                 dynamicExecutionDelegate: dynamicExecutionDelegate,
-                                                 cacheKey: taskKey.cacheKey,
-                                                 jobTaskIDBase: jobTaskIDBase) != 0 {
-                    state = .waitingForOutputDownloads
-                } else {
-                    state = .done
-                }
-            } else {
-                let key = SwiftCachingKeyQueryTaskKey(casOptions: taskKey.casOptions, cacheKey: taskKey.cacheKey, compilerLocation: taskKey.compilerLocation)
+            let compOutputs = try taskKey.cacheKeys.compactMap {
+                try cas.queryLocalCacheKey($0)
+            }
+
+            if compOutputs.count != taskKey.cacheKeys.count {
+                let key = SwiftCachingKeyQueryTaskKey(casOptions: taskKey.casOptions, cacheKeys: taskKey.cacheKeys, compilerLocation: taskKey.compilerLocation)
```

Comment:
Should we only do a remote key query for the keys that were not found locally above?

Reply:
Good catch. I can keep track of the missing keys and only request querying those.